ConfigSpace/hyperparameters.pyx

```diff
@@ -1710,6 +1714,8 @@ cdef class NormalIntegerHyperparameter(IntegerHyperparameter):
 
         self.mu = mu
         self.sigma = sigma
+        if self.sigma == 0:
+            assert isinstance(self.mu, int)
```

Comment:
Why this assertion? I feel like they should both just be `int`. The init signature seems off here to suggest otherwise.

Reply:
This follows from below. But I'm also fine with restoring the existing behavior if allowing a non-integer mean is undesirable.